flocker/node/script.py

```diff
@@ -148,9 +154,18 @@ class ZFSAgentScript(object):
     a Flocker cluster.
     """
     def main(self, reactor, options, volume_service):
-        configuration = agent_config_from_file(path=options[u'agent-config'])
+        try:
+            agent_config = options[u'agent-config']
+            configuration = yaml.safe_load(agent_config.getContent())
+        except IOError:
+            raise ConfigurationError(
+                "Configuration file does not exist at '{}'.".format(
+                    agent_config.path))
+
+        validate_configuration(configuration=configuration)
```

Comment:
It would be good if all of this code were still in a separate function for loading and validating the configuration.  As an extra bonus, that function could also serve as a central point to supply defaults (rather than forcing all code using the configuration to supply per-item defaults in an ad hoc way).  Please file an issue and link to it from here.